### ISSUE TITLE
script to check easyconfig style

### ISSUE THREAD

**Issue by boegel** [easybuilders/easybuild-easyconfigs#1764]
We could really use a script that performs some simple checks on easyconfigs w.r.t. style, which can be used by Jenkins to test incoming easyconfig pull requests and provide some meaningful feedback.

Checks should include (but are not limited to):
- order of parameters
- grouping of parameters
- each parameter should be defined only once
- some non-mandatory yet expected parameters (e.g. moduleclass)
- whitespace (no trailing whitespace, no tabs)
- indentation (may be very tricky)

The script should:
- be easily extendable; adding extra checks should be as simple as implementing a small function like:

``` python
def check_foo(ectxt):
    # check for foo
```
- exit with exit code 0 if all checks pass, exit with non-zero (# failed checks?) otherwise
- provide some meaningful feedback w.r.t. failing checks: which lines fail, what is the problem, how could it be fixed

Maybe also including an auto-restyled easyconfig file (via `EasyConfig.dump()`) could be useful too, by means of inspiration or as a 'good' example of how the easyconfig should look like.

**Comment by boegel** [comment]
@Caylo: please take a look at this

**Comment by boegel** [comment]
Superseded by #2506
